extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/kafka/KafkaStreamsEventObserver.java

```diff
@@ -0,0 +1,66 @@
+package io.quarkus.micrometer.runtime.binder.kafka;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+
+import org.apache.kafka.streams.KafkaStreams;
+import org.jboss.logging.Logger;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.binder.kafka.KafkaStreamsMetrics;
+import io.quarkus.runtime.ShutdownEvent;
+
+/**
+ * Observer to create and register KafkaStreamsMetrics.
+ * 
+ * Must be separated from KafkaEventObserver, because they use different dependencies and if only kafka-client is used, the
+ * classes from kafka-streams aren't loaded.
+ */
+@ApplicationScoped
+public class KafkaStreamsEventObserver {
+
+    private static final Logger log = Logger.getLogger(KafkaStreamsEventObserver.class);
+    MeterRegistry registry = Metrics.globalRegistry;
+    Map<Object, KafkaStreamsMetrics> kafkaStreamsMetrics = new HashMap<>();
+
+    /**
+     * Manage bind/close of KafkaStreamsMetrics for the specified KafkaStreams client.
+     * If kafkaStreams has not been seen before, it will be bound to the
+     * Micrometer registry and instrumented using a Kafka MeterBinder.
+     * If the producer has been seen before, the MeterBinder will be closed.
+     *
+     * @param kafkaStreams Observed KafkaStreams instance
+     */
+    public synchronized void kafkaStreamsCreated(@Observes KafkaStreams kafkaStreams) {
+        KafkaStreamsMetrics metrics = kafkaStreamsMetrics.remove(kafkaStreams);
+        if (metrics == null) {
+            metrics = new KafkaStreamsMetrics(kafkaStreams);
+            try {
+                metrics.bindTo(registry);
+                kafkaStreamsMetrics.put(kafkaStreams, metrics);
```

Comment:
Why is `kafkaStreamsMetrics` a map? It seems we only ever put a single object into it? What is the purpose of this?

Reply:
to verify by instance equality that you don't see more than one -- this is likely due to making kafka streams observer look like the producer/consumer observer (which is dealing with multiple instances).

Reply:
In which case would there be more than one `KafkaStreams` instance?

Reply:
sounds like not, so this can be simplified to just see the event coming in and act accordingly (and then preserve shutdown for cleanup)

Reply:
right, the map is not needed here. I replaced it and adjusted the rest.